### DISCUSSION TITLE
Add markdown removal for TTS

### DISCUSSION THREAD

**Discussion by Shulyaka** [home-assistant/architecture#1120] [Entity Models] [answered]
Problem:
LLM may occasionally output markdown-formatted text even when explicitly told to produce plain text only. When this text is then used with Text-to-speech synthesizers, it may produce unexpected results.
Some TTS engines might actually need the formatting if they can emphasize something, display it, or remove themselves. OpenAI Text-To-Speech is one of the examples according to [this comment](https://community.openai.com/t/tts-add-emphasis-to-one-word-in-spoken-text/706800/10).
However others will translate the markdown formatting into words, such as "underscore", "asterisk", etc.

Suggestion:
1. Allow tts engines to specify that they are able to handle markdown formatting, by adding `markdown` into its `supported_options`.
2. Remove the markdown formatting before sending the text to the tts engine, if the option is not set (which would be the default).

If `supported_options` is not the right place for that (as we are not going to pass this option), we can use `supported_features` instead.

**Comment by einord** [comment]
This would be a very nice feature. I’m currently trying to prevent assist (ChatGPT) to use markdown in the prompt, but doesn’t work all the time.

**Comment by spudje** [comment]
I would love to see this as weel. I can't seem to come up with a prompt that gets rid of those double asterisks for bold text in responses, especially when you enable web search capabilities to the conversation AI.

**Comment by frenck** [comment, marked as answer]
We have discussed this one today in an architectural meeting and concluded that we will decline this proposal.

We believe this is not a concern to solve on the TTS level but on an LLM level. The response's contents are not the responsibility of the TTS entity component. 

../Frenck
